Remove L1 contracts from L2 state dump process

spiyer99 opened the pull request:
**Description**
Removes the l1 only contracts from the deployment configuration since these are not actually deployed using this configuration. 

**Additional context**
Doesn’t change too much. This is my first commit to this repo.

**Metadata**
- Fixes #1191

smartcontracts commented:
Nice. For some context, we used to use this as our L1 and L2 deployment script but have since moved to using hardhat-deploy for L1. Super useful PR!

spiyer99 commented:
@smartcontracts thanks for your help! You made this PR super easy. I noticed that there are 3 workflows awaiting approval, do you need anything from me to get those done?

tynes commented:
Hey @spiyer99, could you add a patch changeset for the contracts package using `yarn changeset` and squash the commits into a single commit that looks like `contracts: remove l1 contracts from l2 state dump process`

spiyer99 commented:
Sure @tynes, will do. I'll reply here when I'm done.

spiyer99 commented:
@tynes, I added a patch changeset for the contracts package and squashed the commits into a single commit using the message you specified earlier. Hopefully I've done this correctly.

smartcontracts commented:
Looking good! Could you just rebase this on top of the latest version of `develop` and then mark it as ready for review?

smartcontracts commented:
`git fetch` followed by `git rebase origin/develop` or `git rebase -i origin/develop` (if you want an interactive rebase)

spiyer99 commented:
Thanks for the help @smartcontracts!

smartcontracts commented on packages/contracts/src/contract-deployment/config.ts:
```diff
@@ -192,59 +105,16 @@ export const makeContractDeployConfig = async (
         config.ovmGlobalContext,
       ],
     },
-    OVM_StateManager: {
```
Whoops my apologies, this contract should actually still be included.

spiyer99 replied:
@smartcontracts no worries, I included the contract in the latest commit.

smartcontracts commented:
Hey! Sorry for the delay here. Would you mind rebasing one more time so we can get this baby merged? Thank you!

spiyer99 commented:
@smartcontracts just rebased it. Let me know if there are any more issues. Also, how else could I help out here?

smartcontracts commented:
Merged! Great job here @spiyer99. If you're looking for an easy issue to tackle, we need to remove this code from the `L1CrossDomainMessenger` and fix any tests that might break when that code gets removed: https://github.com/ethereum-optimism/optimism/blob/029cafce204312c06dc761572599cf3872dd7800/packages/contracts/contracts/L1/messaging/L1CrossDomainMessenger.sol#L78-L95

Please make this change against the `experimental` branch if you decide to tackle it! https://github.com/ethereum-optimism/optimism/tree/experimental

spiyer99 commented:
> Merged! Great job here @spiyer99. If you're looking for an easy issue to tackle, we need to remove this code from the `L1CrossDomainMessenger` and fix any tests that might break when that code gets removed:
> 
> https://github.com/ethereum-optimism/optimism/blob/029cafce204312c06dc761572599cf3872dd7800/packages/contracts/contracts/L1/messaging/L1CrossDomainMessenger.sol#L78-L95
> 
> Please make this change against the `experimental` branch if you decide to tackle it! https://github.com/ethereum-optimism/optimism/tree/experimental

@smartcontracts thanks for merging the PR! 

Happy to work on the issue mentioned above. Should I create a formal issue before submitting a PR?
